tortoise/backends/aioodbc/client.py

```diff
@@ -0,0 +1,277 @@
+import asyncio
+from functools import wraps
+from typing import Any, Callable, List, Optional, SupportsInt, Tuple, TypeVar, Union
+
+import aioodbc
+import pyodbc
+# from asyncpg.transaction import Transaction
+import pypika
+from pypika import OracleQuery as OrigOracleQuery
+
+
+from tortoise.backends.aioodbc.executor import AioodbcExecutor
+from tortoise.backends.aioodbc.schema_generator import AioodbcSchemaGenerator
+from tortoise.backends.base.client import (
+    BaseDBAsyncClient,
+    BaseTransactionWrapper,
+    Capabilities,
+    ConnectionWrapper,
+    NestedTransactionPooledContext,
+    PoolConnectionWrapper,
+    TransactionContext,
+    TransactionContextPooled,
+)
+from tortoise.exceptions import (
+    DBConnectionError,
+    IntegrityError,
+    OperationalError,
+    TransactionManagementError,
+)
+
+FuncType = Callable[..., Any]
+F = TypeVar("F", bound=FuncType)
+
+
+def translate_exceptions(func: F) -> F:
+    @wraps(func)
+    async def translate_exceptions_(self, *args):
+        try:
+            return await func(self, *args)
+        except pyodbc.ProgrammingError as exc:
+            raise OperationalError(exc)
+        except pyodbc.IntegrityError as exc:
+            raise IntegrityError(exc)
+        except pyodbc.Error as exc:
+            raise OperationalError(exc)
+        # except pyodbc.erras exc:  # pragma: nocoverage
+            # raise TransactionManagementError(exc)
+
+    return translate_exceptions_  # type: ignore
+
+
+
+
+
+
+from pypika import OracleQuery as OrigOracleQuery
+from pypika.queries import QueryBuilder, Query
+from pypika import Dialects
+
+
+class OracleQueryBuilder(QueryBuilder):
+    def __init__(self, **kwargs):
+        super(OracleQueryBuilder, self).__init__(dialect=Dialects.ORACLE, **kwargs)
+
+    def get_sql(self, *args, **kwargs):
+        return super(OracleQueryBuilder, self).get_sql(
+            *args, groupby_alias=False, **kwargs
+        )
+
+    def _limit_sql(self):
+        return f' FETCH NEXT {self._limit} ROWS ONLY'
+
+
+class OracleQuery(Query):
+    """
+    Defines a query class for use with Oracle.
+    """
+
+    @classmethod
+    def _builder(cls, **kwargs):
+        builder = OracleQueryBuilder(**kwargs)
+        builder.QUOTE_CHAR = ""
+        return builder
+
+
+
+DSN_TEMPLATE = "ODBC;Driver={{{driver}}};SERVER={c.host}:{c.port}/c.dataabase;UID={c.user};PWD={c.password};SCHEMA={c.schema};"
+
+class AioodbcDBClient(BaseDBAsyncClient):
+    query_class = OracleQuery
+    executor_class = AioodbcExecutor
+    schema_generator = AioodbcSchemaGenerator
+    capabilities = Capabilities("oracle")
```

Comment:
I feel it would be better to call the backend "oracle" instead of aioodbc, because it can be used for many different DBs.
Later we can refactor it a bit to be a little cleaner, so it actually works that way?
E.g. Separate dialect + connector as per #72

Reply:
This makes sense with #72 pending. Since the client itself will just be aioodbc specific, it should be easy to simply give it a different Pypika query builder. However, I've still not built out all the various tests and covered the code, so that may not be entirely accurate.

Reply:
Yes, it is partially composable already, just requires sitting down and finishing it.